python/lsst/meas/extensions/gaap/gaap.py

```diff
@@ -0,0 +1,186 @@
+    scalingFactors = ListField(dtype=float, default=[1.0, 1.15],
+                               doc="List of scale factors to scale the seeing")
+    sigmas = ListField(dtype=float, default=[4.0, 5.0],
+                       doc="List of sigmas for Gaussian apertures")
+    modelPsfMatch = ConfigurableField(doc="PSF Gaussianization Task", target=ModelPsfMatchTask)
+
+    def setDefaults(self) -> None:
+        # TODO: The following will move to a config file later in DM-27482
+        self.modelPsfMatch.kernel.active.alardNGauss = 1
+        self.modelPsfMatch.kernel.active.alardDegGaussDeconv = 1
+        self.modelPsfMatch.kernel.active.alardDegGauss = [8]
+        self.modelPsfMatch.kernel.active.alardGaussBeta = 1.0
+        self.modelPsfMatch.kernel.active.spatialKernelOrder = 0
+
+
+class BaseGaapFluxPlugin(GenericPlugin):
+
+    ConfigClass = BaseGaapFluxConfig
+
+    @classmethod
+    def getExecutionOrder(cls) -> float:
+        return cls.FLUX_ORDER
+
+    def getGaapResultName(self, sF: float, sigma: float) -> str:
+        return "_".join((self.name, str(sF).replace(".", "_")+"x", str(sigma).replace(".", "_")))
+
+    def __init__(self, config, name, schema, metadata) -> None:
+        """Gaussian Aperture and PSF (GAaP) photometry plugin.
+        """
+        GenericPlugin.__init__(self, config, name, schema, metadata)
+        for sF, sigma in itertools.product(self.config.scalingFactors, self.config.sigmas):
+            baseName = self.getGaapResultName(sF, sigma)
+            schema.addField(schema.join(baseName, "instFlux"), type="F", doc="GAaP Flux")
+            schema.addField(schema.join(baseName, "instFluxErr"), type="F", doc="GAaP Flux error")
+
+    def convolve(self, exposure: afwImage.Exposure, modelPsf: afwDetection.GaussianPsf,
+                 footprint: afwDetection.Footprint) -> afwImage.Exposure:
+        """Convolve the ``exposure`` to make the PSF same as ``modelPsf``.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Original (full) exposure containing all the sources.
+        modelPsf : `lsst.afw.detection.GaussianPsf`
+            Target PSF to which to match.
+        footprint : `lsst.afw.detection.Footprint`
+            Footprint of the source to measure.
+
+        Returns
+        -------
+        convExp : `lsst.afw.image.Exposure`
+            Sub-image containing the source, convolving to the target seeing
+        """
+
+        bbox = footprint.getBBox()
+        pixToGrow = self.config.modelPsfMatch.kernel.active.detectionConfig.fpGrowPix
+        # This is conservative for now.
+        # An alternate option is 2*max(self.psfMatch.kConfig.sizeCellX, self.psfMatch.kConfig.sizeCellY)
+        bbox.grow(pixToGrow)  # gets too big
+
+        origPsf = exposure.getPsf()
+
+        maskedImage = exposure.getMaskedImage()
+        subImage = maskedImage.Factory(maskedImage, bbox)
```

Comment:
The syntax was from an earlier bootcamp note of yours on how to write a Plugin. The __getitem__ syntax didn't work back then since it did not pay attention to xy0 and found it in at least one plugin that was created after that.

I did observe failures for sources near the edges when running on RC2 and have created a ticket (DM-28925) to fix it.

Reply:
:+1:  That problem with `__getitem__` not paying attention to x0 should now be fixed, so it's safe to use now.